typescript/radix-sdk/src/ism/routing-ism.ts

```diff
@@ -0,0 +1,236 @@
+import { GatewayApiClient } from '@radixdlt/babylon-gateway-api-sdk';
+
+import { IProvider, ISigner, IsmType } from '@hyperlane-xyz/provider-sdk/altvm';
+import { ChainLookup } from '@hyperlane-xyz/provider-sdk/chain';
+import {
+  DerivedIsmConfig,
+  DomainRoutingIsmConfig,
+  ExtractIsmModuleType,
+  IsmConfig,
+  IsmModuleAddresses,
+  IsmModuleType,
+  calculateDomainRoutingIsmDelta,
+  extractIsmAddress,
+} from '@hyperlane-xyz/provider-sdk/ism';
+import {
+  AnnotatedTx,
+  HypModule,
+  HypModuleArgs,
+  HypReader,
+  ModuleProvider,
+  ReaderProvider,
+  TxReceipt,
+} from '@hyperlane-xyz/provider-sdk/module';
+import { WithAddress, assert, eqAddressRadix } from '@hyperlane-xyz/utils';
+
+import { AnnotatedRadixTransaction } from '../utils/types.js';
+
+import { getDomainRoutingIsmConfig } from './query.js';
+import { RadixRoutingIsmTx } from './tx.js';
+
+type RoutingIsmModule = ExtractIsmModuleType<'domainRoutingIsm'>;
+
+export class RadixRoutingIsmReader implements HypReader<RoutingIsmModule> {
+  constructor(
+    private readonly provider: IProvider,
+    private readonly radixGateway: GatewayApiClient,
+    private readonly readerProvider: ReaderProvider<IsmModuleType>,
+  ) {}
+
+  async read(address: string): Promise<WithAddress<DomainRoutingIsmConfig>> {
+    const { owner, routes } = await getDomainRoutingIsmConfig(
+      this.radixGateway,
+      address,
+    );
+
+    const ismReader = this.readerProvider.connectReader(this.provider);
+
+    const domains: Record<string, DerivedIsmConfig> = {};
+    for (const route of routes) {
```

Comment:
For the record (we discussed this outside Github): let's not do any nested operations on the provider level. We should keep all compound logic (ISMs nested within ISMs in this case) in deploy-sdk.